src/random/random.c

```diff
@@ -228,27 +255,54 @@ void igraph_rng_destroy(igraph_rng_t *rng) {
  * Time complexity: usually O(1), but may depend on the type of the
  * RNG.
  */
-igraph_error_t igraph_rng_seed(igraph_rng_t *rng, unsigned long int seed) {
+igraph_error_t igraph_rng_seed(igraph_rng_t *rng, igraph_uint_t seed) {
     const igraph_rng_type_t *type = rng->type;
-    rng->def = 0;
+    rng->is_seeded = 1;
```

Comment:
Nitpick: I'd move this line after the next, so that in case the next line fails, `is_seeded` stays false.